homeassistant/components/media_player/soundtouch.py

```diff
@@ -297,7 +297,7 @@ def media_album_name(self):
     def play_media(self, media_type, media_id, **kwargs):
         """Play a piece of media."""
         _LOGGER.debug("Starting media with media_id: %s", media_id)
-        if re.match(r'http://', str(media_id)):
+        if re.match(r'https?://', str(media_id)):
```

Comment:
You should compile this statement on top.
```
URL = re.compile(r"..")
...
if URL.match(media_id):
```

Reply:
compiling regexes is generally a best practice, but as this is a function which will be called rarely it seems like unnecessary overhead here to compile a regex for just one execution no?

Reply:
Yes. You call this regex every time where some one play a URL with this component. Maybe you use it 1-2 in a month but other could use it 3-6 per day

Reply:
@pvizeli If we create a URL object here, it will exist for that one match, and as soon as the `play_media` function is completed, it will go out of scope, and be scrapped... the regular expression is used exactly one time per call of the function, and so there is no advantage in compiling it.  Compiling regular expressions is only an advantage when they are being used many times within a context.

Reply:
I think it's fine to just do it here.